src/errors/IsomerError.ts

```diff
@@ -20,26 +7,75 @@ export const ComponentTypes = {
   Model: "M",
   Util: "U",
   Other: "X",
-}
+} as const
+
+export type PossibleComponentTypes = typeof ComponentTypes[keyof typeof ComponentTypes]
 
 export const FileCodes = {
   CollectionPageService: "001",
   Undefined: "000",
+} as const
+
+export type PossibleFileCodes = typeof FileCodes[keyof typeof FileCodes]
+
+type ExternalCode = `${PossibleComponentTypes}-${PossibleFileCodes}-${string}`
+
+/* External representation of Isomer errors - passed to frontend client */
+export interface IsomerExternalError {
+  code: ExternalCode
+  message: string
+}
+
+/* Internal representation of Isomer errors - passed to log stream */
+export interface IsomerInternalError {
+  code: string
+  message: string
+  meta?: Record<string, unknown>
+}
+
+export interface IdentifiableError {
+  componentCode: PossibleComponentTypes
+  fileCode: PossibleFileCodes
 }
 
 const getExternalCode = (
-  componentCode: string,
-  fileCode: string,
+  componentCode: PossibleComponentTypes,
+  fileCode: PossibleFileCodes,
   errCode: string
-): string => `${componentCode}-${fileCode}-${errCode}`
+): ExternalCode => `${componentCode}-${fileCode}-${errCode}`
 
 /* Namespace for Error functions */
 export const IsomerError = {
-  toExternalRepresentation: (e: IsomerInternalError): IsomerExternalError => ({
+  toExternalRepresentation: <E extends IsomerInternalError & IdentifiableError>(
+    e: E
+  ): IsomerExternalError => ({
     code: getExternalCode(e.componentCode, e.fileCode, e.code),
     message: e.message,
   }),
   getLog: (error: IsomerInternalError): Record<string, unknown> => ({
     ...error,
   }),
+  isIdentifiable: <E extends BaseIsomerError>(
+    err: E
+  ): err is E & IdentifiableError =>
+    ((err as unknown) as IdentifiableError).componentCode !== undefined &&
+    ((err as unknown) as IdentifiableError).fileCode !== undefined,
+}
+
+interface LegacyRepresentation {
```

Comment:
```suggestion
interface LegacyErrorRepresentationInputs {
```

Reply:
would prefer to have it as `LegacyErrorRepresentation` - this is not really an `Input` more an interface that existing errors fulfill.